src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -76,15 +87,50 @@ class MoneyRequestAmountPage extends React.Component {
     }
 
     componentDidMount() {
+        if (this.isEditing) {
+            if (_.isEmpty(this.props.moneyRequest.participants) || this.props.moneyRequest.amount === 0) {
+                Navigation.goBack();
+                Navigation.navigate(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID));
+            }
+        } else {
+            // Set the money request participants based on the report participants
+            const participants = ReportUtils.isPolicyExpenseChat(this.props.report)
+                ? OptionsListUtils.getPolicyExpenseReportOptions(this.props.report)
+                : OptionsListUtils.getParticipantsOptions(this.props.report, this.props.personalDetails);
+            this.props.moneyRequest.setParticipants(_.map(participants, (participant) => ({...participant, selected: true})));
+        }
+
         this.focusTextInput();
 
         // Focus automatically after navigating back from currency selector
         this.unsubscribeNavFocus = this.props.navigation.addListener('focus', () => {
             this.focusTextInput();
-            this.getCurrencyFromRouteParams();
         });
     }
 
+    componentDidUpdate(prevProps) {
+        const prevCurrencyParam = lodashGet(prevProps.route.params, 'currency', '');
+        const currencyParam = lodashGet(this.props.route.params, 'currency', '');
+        if (currencyParam !== '' && prevCurrencyParam !== currencyParam) {
+            this.setState({selectedCurrencyCode: currencyParam});
+        }
+
+        if (prevProps.moneyRequest.currency !== this.props.moneyRequest.currency) {
+            this.setState({selectedCurrencyCode: this.props.moneyRequest.currency});
+        }
+
+        if (prevProps.moneyRequest.amount !== this.props.moneyRequest.amount) {
+            const selectedAmountAsString = CurrencyUtils.convertToWholeUnit(this.props.moneyRequest.currency, this.props.moneyRequest.amount).toString();
+            this.setState({
+                amount: selectedAmountAsString,
+                selection: {
+                    start: selectedAmountAsString.length,
+                    end: selectedAmountAsString.length,
+                },
+            });
+        }
+    }
```

Comment:
For instance the currency in `selectedAmountAsString` may on be correct. e.g. we'd be using currency from `moneyRequest` to calculate amount while the currency set in state is the currency from param.

Reply:
I don't think there will be the case like you mentioned.

This is to update the currency after we select it in currency selection page, so amount will stay the same.
```
const prevCurrencyParam = lodashGet(prevProps.route.params, 'currency', '');
const currencyParam = lodashGet(this.props.route.params, 'currency', '');
if (currencyParam !== '' && prevCurrencyParam !== currencyParam) {
    this.setState({selectedCurrencyCode: currencyParam});
}
```

This is to update the currency when we update it in edit amount page.
```
if (prevProps.moneyRequest.currency !== this.props.moneyRequest.currency) {
    this.setState({selectedCurrencyCode: this.props.moneyRequest.currency});
}
```

This is to update the amount when we update it in edit amount page.
```
if (prevProps.moneyRequest.amount !== this.props.moneyRequest.amount) {
    const selectedAmountAsString = CurrencyUtils.convertToWholeUnit(this.props.moneyRequest.currency, this.props.moneyRequest.amount).toString();
    this.setState({
        amount: selectedAmountAsString,
        selection: {
            start: selectedAmountAsString.length,
            end: selectedAmountAsString.length,
        },
    });
}
```

In case both currency and amount get updated, there won't be a problem as we use the currency props to calculate the amount as string. Why is it important to call `convertToWholeUnit` only every time the amount changes? This is because when we save the amount, it will call `convertToSmallestUnit` (it will also save the currency from state to context). I don't really understand the meaning from the name, but here is an example of USD currency.

Let say we input 25, `convertToSmallestUnit` will convert it to 2500 (and this is the one that get saved in the `amount` context) and then `convertToWholeUnit` will convert it back to 25.

Reply:
I guess you are right.